pkgs/development/python-modules/annoy/default.nix

```diff
@@ -13,6 +14,8 @@ buildPythonPackage rec {
     sha256 = "fe2779664bd8846f2d67191a7e6010b8df890ac4586336748fd0697f31654379";
   };
 
+  nativeBuildInputs = [ h5py ];
```

Comment:
this is typically a run-time dependency

Reply:
what I thought too:
```
$ ls ./result/lib/python3.7/site-packages/annoy/
annoylib.cpython-37m-x86_64-linux-gnu.so  __init__.py
$ ldd ./result/lib/python3.7/site-packages/annoy/annoylib.cpython-37m-x86_64-linux-gnu.so
	linux-vdso.so.1 (0x00007ffd6d7ff000)
	libpython3.7m.so.1.0 => /nix/store/ga9kam5sgryz01frprfiy0b0kblmy1wi-python3-3.7.6/lib/libpython3.7m.so.1.0 (0x00007fde835dd000)
	libstdc++.so.6 => /nix/store/3cngcg3w6gvck6ngj60byx843f27pf76-gcc-9.2.0-lib/lib/libstdc++.so.6 (0x00007fde833fd000)
	libm.so.6 => /nix/store/4zhy7p7dhfjbaac0i0kb6f2ky1w3747k-glibc-2.27/lib/libm.so.6 (0x00007fde83267000)
	libgcc_s.so.1 => /nix/store/4zhy7p7dhfjbaac0i0kb6f2ky1w3747k-glibc-2.27/lib/libgcc_s.so.1 (0x00007fde83051000)
	libpthread.so.0 => /nix/store/4zhy7p7dhfjbaac0i0kb6f2ky1w3747k-glibc-2.27/lib/libpthread.so.0 (0x00007fde83030000)
	libc.so.6 => /nix/store/4zhy7p7dhfjbaac0i0kb6f2ky1w3747k-glibc-2.27/lib/libc.so.6 (0x00007fde82e78000)
	libdl.so.2 => /nix/store/4zhy7p7dhfjbaac0i0kb6f2ky1w3747k-glibc-2.27/lib/libdl.so.2 (0x00007fde82e73000)
	libcrypt.so.1 => /nix/store/4zhy7p7dhfjbaac0i0kb6f2ky1w3747k-glibc-2.27/lib/libcrypt.so.1 (0x00007fde82e39000)
	libncursesw.so.6 => /nix/store/1m5v2mmbvmqb9gm8m8zq77zf2pf3h9nh-ncurses-6.1-20190112/lib/libncursesw.so.6 (0x00007fde82dc7000)
	libutil.so.1 => /nix/store/4zhy7p7dhfjbaac0i0kb6f2ky1w3747k-glibc-2.27/lib/libutil.so.1 (0x00007fde82dc2000)
	/nix/store/xhpwab5kavygbr1fswawmdyqvmn3wa4i-glibc-2.27/lib64/ld-linux-x86-64.so.2 (0x00007fde8394c000)
```

Reply:
i guess since it's linked against libdl, it could also dlopen h5py, wish this had useful tests :(